src/MatrixFields/field_name_dict.jl

```diff
@@ -237,6 +334,251 @@ function Base.one(matrix::FieldMatrix)
     return FieldNameDict(inferred_diagonal_keys, entries)
 end
 
+"""
+    field_offset_and_type(name_pair::FieldNamePair, ::Type{T}, ::Type{S}, full_key::FieldNamePair)
+
+Returns the offset of the field with name `name_pair` in an object of type `S` in
+multiples of `sizeof(T)`, the type of the field with name `name_pair`, and a `Val` indicating
+what method can index a ClimaCore `Field` of `S` with `name_pair`.
+
+The third return value is one of the following:
+- `Val(:view)`: indexing with a view is possible
+-  `Val(:view_of_blocks)`: indexing with a view of blocks is possible (this is not implemented)
```

Comment:
Could you add a line to clarify this? I'm interpreting this to mean that a `view_of_blocks` would involve a strided view with nonuniform stride length, but I'm not sure if I got that right.